src/odemis/util/test/img_test.py

```diff
@@ -1181,8 +1183,873 @@ def test_rotate_img_metadata(self):
+        """
+
+        size = (2000, 1000)
+        md = {
+              model.MD_DIMS: 'YX',
+              model.MD_PIXEL_SIZE: (1e-6, 1e-6),
+              model.MD_BPP: 0
+             }
+
+        # check for dtype uint8
+        arr = numpy.arange(size[0] * size[1], dtype=numpy.uint8)
+        data = model.DataArray(arr, metadata=md)
+
+        # min/max values should be uint8 min/max
+        min_max_type = numpy.iinfo(numpy.uint8)
+
+        # raises the ValueError as well
+        min_max_values = img.guessDRange(data)
+
+        self.assertEqual(min_max_values[0] == min_max_type.min, min_max_values[1] == min_max_type.max)
+
+    def test_guess_drange_wrong_type(self):
+        """
+        Test if an unexpected array datatype will raise a TypeError
+        """
+
+        size = (2000, 2000)
+        md = {
+              model.MD_DIMS: 'YX',
+              model.MD_PIXEL_SIZE: (1e-6, 1e-6),
+              model.MD_BPP: 8
+             }
+
+        # check for dtype float (=wrong type)
+        arr = numpy.arange(size[0] * size[1], dtype=numpy.float)
+        data = model.DataArray(arr, metadata=md)
+
+        with self.assertRaises(TypeError):
+            img.guessDRange(data)
+
+    def test_guess_drange_normal(self):
+        """
+        Test the data range of the data based on guess method with regular
+        model dict, type and DA values
+        """
+
+        size = (4000, 4000)
+        md = {
+              model.MD_DIMS: 'YX',
+              model.MD_PIXEL_SIZE: (1e-6, 1e-6),
+              model.MD_BPP: 8
+             }
+
+        # check for dtype int32
+        arr = numpy.arange(size[0] * size[1], dtype='int32')
+        data = model.DataArray(arr, metadata=md)
+
+        # min/max values are dependent on given BPP
+        min_max_values = img.guessDRange(data)
+
+        self.assertEqual(min_max_values[0] == -abs(2 ** data.metadata[model.MD_BPP] // 2),
+                         min_max_values[1] == (2 ** data.metadata[model.MD_BPP] // 2) - 1)
+
+
+class TestGetColorBar(unittest.TestCase):
+    # input matplotlib colormap, width int, height int and alpha channel bool
+    # output numpy Array of uint8 RGB tuples
+    # test for array shape => tuple 3 values
+    # test for dtype uint8
+    # test for RGB value range
+
+    def test_get_colorbar_alpha(self):
+        """
+        Test if input of a registered matplotlib.colors colormap will return a ndArray of
+        the right tuple shape, within ranges of uint8 and with alpha channel included
+        """
+
+        # input will be color_map (obj matplotlib) height (int), width (int) and alpha (bool)
+        input_cm = cm.get_cmap('cividis')
+        input_height = 16
+        input_width = 72
+        input_alpha = True
+
+        # NumPy array of uint8 RGB tuples is returned
+        output_array = img.getColorbar(input_cm, input_height, input_width, input_alpha)
+
+        # test for tuple shape of 3
+        self.assertEqual(len(output_array.shape), 3)
```

Comment:
You have much strong expectations, lets check for them: shape == (input_height, input_width, 4)